### DISCUSSION TITLE
IS this supported?

### DISCUSSION THREAD

**Discussion by ionultd** [Koenkk/zigbee2mqtt#7414] [Devices]
I have recently purchased a Zigbee 3.0 USB dongle from itead which I found out uses the EFR32MG21 chip instead of the CC2531. Is this chip supported by zigbee2mqtt?

Thanks

**Comment by kirovilya** [comment]
Basically, read first post here https://github.com/Koenkk/zigbee-herdsman/issues/319

**Reply by Hedda** [reply]
FYI, many users and developers from different open-source home automation applications have reported that the first initial hardware revision (still the current hardware version) of this Zigbee 3.0 dongle from ITead is seriously flawed because it features no electromagnetic shielding as well as use an antenna design that is very sensitive to electromagnetic interference / radio frequency interference as those caused by example Wi-Fi Access Points and electronic computers/appliances//devices/accessories/wires/cables/power-supplies that do not have properly shielded electronics.

Among other problem symptoms issues with signal interference can be seen as problems pairing new Zigbee devices unless they are either very close or you use a (very) long USB extension cable to get the dongle/stick far away from your computer and any other devices/appliances that can be a source of electromagnetic interference. Getting the dongle/stick away from other electronic devices/appliances have proven to resolve Zigbee pairing issues and been seen to significantly improve RSSI (Received Signal Strength Indicator) and LQI (Link Quality Indicator), and PRR (Packet Reception Ratio) metrics.

A suggested workaround other than using a long USB extension cable or moving away from sources of interference is to add your own electromagnetic shielding by first wrapping everything on the dongles/sticks circuit board except the PCB trace antenna part with a layer of isolating tape/firm and then adding a layer of conductive metallic/copper tape on top of that while making sure the metal shielding touches the USB plug jacketing for grounding but not covering the antenna part of the board.

https://community.home-assistant.io/t/itead-zigbee-3-0-usb-dongle-stick-only-cost-7-and-is-based-on-silicon-labs-efr32mg21/271144/44

https://community.openhab.org/t/itead-zigbee-3-0-usb-dongle-stick-only-cost-7-and-is-based-on-silicon-labs-efr32mg21/115034/128

**Reply by ionultd** [reply]
Thanks for that additional info but I see I need to be more specific. I cannot even get the dongle connected to HASS. I plug the dongle into my ubuntu 20.04 host and then try to find it using my HASS VM and the zigbee2mqtt addon but it does not appear to detect it on any USB port.

**Reply by Hedda** [reply]
> Thanks for that additional info but I see I need to be more specific. I cannot even get the dongle connected to HASS. I plug the dongle into my ubuntu 20.04 host and then try to find it using my HASS VM and the zigbee2mqtt addon but it does not appear to detect it on any USB port.

Then sounds like a serial device path issue in operating system (Home Assistant OS) or Docker and not a Zigbee2MQTT issue.

I read there were issues with serial in Home Assistant 2021.5.x later reported as fixed in 2021.5.5

https://github.com/home-assistant/core/issues/50912

https://github.com/home-assistant/core/issues/50289

One workaround for Home Assistant 2021.5.x could be to use /dev/ttyUSB0:/dev/ttyUSB0 instead of /dev/serial/by-id/xxxyyy

**Reply by Hedda** [reply]
Oh, and if using it in a VM then you of course need to give the VM exclusive right to that USB port and/or serial device.

Try searching for keywords that include "virtual machine" AND "zigbee" for more tips on getting adapter to be seen.

**Comment by Hedda** [comment]
Slightly off-topic but FYI; Zigbee Router firmware for EFR32MG21 adapters has now been released by xsp1989 on GitHub.

https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle/RouterForDongle

From readme it sounds router firmware tested with both the ITead Zigbee 3.0 USB Dongle and an SM-011 based USB adapter.

https://github.com/xsp1989/zigbeeFirmware/blob/710a60451f6631cad46f5d5107b65198558709c3/firmware/Zigbee3.0_Dongle/RouterForDongle/README.md

The same “SM-011 V1.0” Zigbee radio module design by CoolKit Technologies is also used inside some Zigbee gateways/hubs like the popular ITead Sonoff ZBBridge and ZB-GW03 eWeLink Ethernet Zigbee Gateway sold by EACHEN and SmartWise, so could perhaps be that the same Zigbee Router firmware could maybe also be used on the SM-011 Zigbee module in all or some of those products as well?

On-topic discussions here:

https://github.com/xsp1989/zigbeeFirmware/issues/16

https://github.com/xsp1989/zigbeeFirmware/issues/2

**Comment by Hedda** [comment]
Note! EZSP support is still in a very experimental phase so you should really post your feedback here -> https://github.com/Koenkk/zigbee-herdsman/issues/319

I am not sure if developer(s) is currently actively working on improving or maintaining that experimental support(?).

Only clear tips is to try upgrade EmberZNet NCP UART EZSP firmware Zigbee firmware version 6.7.9 or later and test dev branch (a.k.a. Edge):

https://github.com/arendst/Tasmota/tree/development/tools/fw_SonoffZigbeeBridge_ezsp

https://github.com/Koenkk/zigbee2mqtt/tree/dev

There looks to be some ezsp related fixes in the zigbee-herdsman dependency that might have not yet made it into Z2M dev/edge yet:

https://github.com/Koenkk/zigbee-herdsman/

Regardless you still want to direct most if not all feedback and issues to subscribers of the WIP discussion in the zigbee-herdsman repo:

https://github.com/Koenkk/zigbee-herdsman/issues/319

Also, please take note that WiFi based Zigbee bridges like ITead Sonoff ZBBridge are generally not recommended with the EZSP protocol. See the warnings about it:

https://www.zigbee2mqtt.io/guide/adapters/#notes

WiFi-based Serial-to-IP bridges are not recommended for Silicon Labs EZSP adapters as the serial protocol does not have enough fault-tolerance to handle packet loss or latency delays that can normally occur over WiFi connections. If cannot use a locally connected USB or UART/GPIO adapter then the recommendation is to use remote adapter that connected via Ethernet (wired) to avoid issues with EZSP caused by WiFi connections.

https://www.zigbee2mqtt.io/advanced/remote-adapter/connect_to_a_remote_adapter.html

WiFi-based Serial-to-IP bridges are not recommended for Silicon Labs EZSP adapters as the serial protocol does not have enough fault-tolerance to handle packet loss or latency delays that can normally occur over WiFi connections.

More warning about Zigbee to WiFi bridges for EZSP is not recommended for production as it does not do well with packed loss:

https://github.com/zigpy/bellows#warning-about-zigbee-to-wifi-bridges

https://www.home-assistant.io/integrations/zha/#warning-about-wi-fi-based-zigbee-to-serial-bridgesgateways

FYI, I also understand that the EFR32 board antenna design of the "SM-011 V1.0" Zigbee module used inside ITead's Sonoff ZBBridge have very high receive sensitivity to electromagnetic interference (EMI) / radio-frequency interference. This means that it will be extra important to keep it away from any other electronic devices or appliances that may generate an electromagnetic field (EM Field) or send our interfering radio signals. That includes the computer that you run your home automation application software on as well as any devices such as for example external harddrives or storage devices, power-supplies, etc., and of course any Wi-Fi access points or routers nearby.

Note! The new ITead Zigbee 3.0 USB Dongle Model: 9888010100045 (Hardware Revision Version 1.3) circuit board share the same type of antenna design so its radio is as well extremely sensitive to signal interference on its receiving end. The current recommendation for using it is to connect it via a long USB extension cable and try to move it as far away from all electronics and other possible sources of signal interference.

PS: Another general tip that always applies when building a Zigbee network is to begin by adding a few good mains-powered Zigbee router devices so that they will make use a stable baseline. Many start by buying three "IKEA TRADFRI Signal Repeater" (E1746).

https://zigbee.blakadder.com/Ikea_E1746.html

https://www.ikea.com/us/en/p/tradfri-signal-repeater-30400407/

https://www.ikea.com/gb/en/p/tradfri-signal-repeater-80424255/

https://www.ikea.com/se/sv/p/tradfri-signalfoerstaerkare-10400408/
